framework/src/utils/MooseTypes.C

```diff
@@ -25,14 +25,15 @@ namespace Moose
 const std::vector<ExecFlagType>
 populateExecTypes()
 {
-  std::vector<ExecFlagType> exec_types(7);
+  std::vector<ExecFlagType> exec_types(8);
   exec_types[0] = EXEC_INITIAL;
   exec_types[1] = EXEC_TIMESTEP_BEGIN;
   exec_types[2] = EXEC_NONLINEAR;
   exec_types[3] = EXEC_LINEAR;
   exec_types[4] = EXEC_TIMESTEP_END;
   exec_types[5] = EXEC_CUSTOM;
   exec_types[6] = EXEC_SUBDOMAIN;
+  exec_types[7] = EXEC_FINAL;
```

Comment:
I looked at this before and unfortunately this exec_types thing is spread throughout MOOSE in several files. You've managed to hack it in just enough here, but you've left it pretty incomplete. You only added it to one Executioner, we already have a execute on final in there for Outputs so this is somewhat inconsistent with the way you've added it here.

Reply:
I'm not sure what you mean about already having execute on final for outputs.  The execute on final I added was to both steady and transient executioners.  The PerflogDumper was not running on final until I made the changes here.  I can certainly take a look around to see what other places I may have missed.

Reply:
I know we had exec_types or other things spread all over the place. Lots of inconsistencies. Maybe most of it was cleaned up here (https://github.com/idaholab/moose/pull/8946) but I wonder if your new addition is everything we need, that's all I was wondering.

Reply:
I think I got all the necessary changes made.